diff -r c5e178adb138 USBDevice/USBHAL_Maxim.cpp
--- a/USBDevice/USBHAL_Maxim.cpp	Fri Nov 11 17:59:00 2016 +0000
+++ b/USBDevice/USBHAL_Maxim.cpp	Thu Mar 30 11:14:58 2017 +0300
@@ -141,8 +141,13 @@
 
     // attach IRQ handler and enable interrupts
     instance = this;
+#if defined(TARGET_MAX32630)
     NVIC_SetVector(USB_IRQn, &_usbisr);
+#else
+    NVIC_SetVector(USB_IRQn, (uint32_t)&_usbisr);
+#endif
     NVIC_EnableIRQ(USB_IRQn);
+
 }
 
 USBHAL::~USBHAL(void)
